### PR TITLE
docs: add testing instructions in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cjdcordeiro**
# Proposed changes

Ensure the Spread testing instructions are part of the contributing guidelines, and that the PR description template is also updated accordingly.

## Related issues/PRs
#281 
#290 
#289 

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -17,22 +17,14 @@ being proposed in this PR (i.e. feat, test, fix, ci, chore, docs).
...
-#  sudo chroot <path> <app>
-```
 
 ## Checklist
 <!-- Go over all the following points, and put an `x` in all the boxes
 that apply. -->
 
 * [ ] I have read the [contributing guidelines](
 https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
+* [ ] I have tested my changes
```
Comment (rebornplusplus) [inline .github/pull_request_template.md:27]:
```suggestion
* [ ] I have tested my changes ([How to](https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md))
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -265,12 +264,41 @@ conflicts between slice definitions files;
 Apart from asserting your slice definitions' formatting, you must also test them
 before opening a Pull Request.
 
-Chisel supports custom releases, so please make sure you `cut` your custom
-slices and run smoke tests on them.
-
-***Tip**: to install slices from a custom Chisel release, simply provide its
-absolute path to the `--release` option in Chisel. E.g.
-`chisel cut --release $PWD/custom-chisel-release --root testfs pkg-a_sliceFoo`.*
+Every Chisel release support the functional testing for slices via
```
Comment (rebornplusplus) [inline CONTRIBUTING.md:267]:
```suggestion
Every Chisel release supports the functional testing for slices via
```
nitpick

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -265,12 +264,41 @@ conflicts between slice definitions files;
...
+
+**We expect tests to be provided for every slice**, so please ensure you include
+those in your PRs.
+
+Here's how you can create and run tests for your slice definitions:
+
+1. under `tests/spread/integration/`, make sure at least one folder matching the
+package name you're slicing, exists,
+1. inside that folder you must create a `task.yaml` file. This is where you'll
+write all the tests for your slide definitions. Please follow the structure from
```
Comment (rebornplusplus) [inline CONTRIBUTING.md:278]: Today I learnt: You can not number and just put 1 in an ordered list and markdown will take care of it!

**Review by rebornplusplus — COMMENTED** [review]
Looks OK to me for the most part. Just a bunch of nitpicks below.

File: .github/pull_request_template.md (hunk 1 above)
Comment (rebornplusplus) [reply .github/pull_request_template.md:27]: Hey, I see you edited the suggestion but didn't apply them.

Comment (cjdcordeiro) [reply .github/pull_request_template.md]: dunno what happened. committed one by one now

File: .github/workflows/spread.yaml
Comment (zhijie-yang) [inline .github/workflows/spread.yaml:1]: Is this duplicating #294 ?

**Review by zhijie-yang — APPROVED** [review]
LGTM. Just one question:

File: .github/workflows/spread.yaml
Comment (cjdcordeiro) [reply .github/workflows/spread.yaml:1]: it was just missing a rebase

**Pull request merged**